### PR TITLE
Add ContractDependencyPaths for ApiCompat

### PR DESCRIPTION
**Pull request by utpilla**
## Changes
- Add ContractDependencyPaths which is required for ApiCompat to work correctly

**Note:** This assumes that both contract (1.0.1) and the implementation (the current binaries generated) have the same dependencies. We will have to update this if we update any dependency of the implementation

### CODE REVIEW

**Review by reyang — APPROVED** [review]
LGTM 👍

**Review by cijothomas — APPROVED** [review]

File: build/Common.prod.props [hunk 1]
```diff
@@ -10,6 +10,10 @@
       <PrivateAssets>All</PrivateAssets>
     </PackageReference>
   </ItemGroup>
+
+  <PropertyGroup Condition="'$(MinVerTagPrefix)' == 'core-' AND '$(CheckAPICompatibility)' == 'true'">
+    <ApiCompatExcludeAttributeList>$(RepoRoot)\build\ApiCompatExcludeAttributeList.txt</ApiCompatExcludeAttributeList>
```
Comment (reyang) [inline build/Common.prod.props:15]: This has to be used somewhere (e.g. passed in as `--exclude-attributes-list`)?

Comment (reyang) [reply build/Common.prod.props:15]: https://github.com/dotnet/wpf/blob/5695fbe1b5767bc318f6630d2245b5376be8d779/eng/WpfArcadeSdk/tools/GenApi.props#L31

Comment (utpilla) [reply build/Common.prod.props:15]: ApiCompat should take the contents of the file and use that as the parameter value for `--exclude-attributes-list`: https://github.com/dotnet/arcade/blob/f925a9e0fad79e87dd1c46205b252c12b685cfa0/src/Microsoft.DotNet.ApiCompat/src/build/Microsoft.DotNet.ApiCompat.targets#L68

I followed this for reference: https://github.com/Azure/azure-sdk-for-net/blob/42d70236d406451f0323fa4de31ee1dab403d2aa/eng/Directory.Build.Common.targets#L244

Comment (reyang) [reply build/Common.prod.props:15]: I see.
Perhaps take a look at the process creation and see what arguments are being used? https://github.com/dotnet/arcade/blob/f925a9e0fad79e87dd1c46205b252c12b685cfa0/src/Microsoft.DotNet.ApiCompat/src/build/Microsoft.DotNet.ApiCompat.targets#L82

**Pull request merged**